feat(bitbucket): main branch status for mend app dashboard

zharinov opened the pull request:
<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->
<!-- Please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md before you create your pull request.-->

## Changes

- Awaits: #38084
- Extends: #32165 (with unit tests)

## Context

Please select one of the below:

- [ ] This closes an existing Issue: #
- [x] This doesn't close an Issue, but I accept the risk that this PR may be closed if maintainers disagree with its opening or implementation

## AI assistance disclosure

<!-- We request this information to assist reviewers in identifying AI-generated errors and other issues specific to AI usage. While we typically permit the use of AI tools, we appreciate being notified when they are employed. -->

Did you use AI tools to create any part of this pull request?

Please select one option and, if yes, briefly describe how AI was used (e.g., code, tests, docs) and which tool(s) you used.

- [ ] No — I did not use AI for this contribution.
- [ ] Yes — minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [x] Yes — substantive assistance (AI generated non‑trivial portions of code, tests, or documentation).
- [ ] Yes — other (please describe):

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

The public repository: <URL>

<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/main/.github/pull_request_template.md -->

<!-- Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. -->
<!-- PRs will always be squashed by us when we merge your work. Commit as many times as you need in this branch. -->

setchy committed:
feat(bitbucket): main branch status for mend app dashboard

zharinov committed:
Merge branch 'main' into feat/bitbucket-mend-dashboard-check

zharinov committed:
feat(bitbucket): Better cache invalidation for branch status cache

zharinov committed:
Merge branch 'feat/bitbucket-mend-dashboard-check' into feat/bitbucket-mend-dashboard-check-ext

zharinov committed:
Add tests

zharinov committed:
Merge branch 'main' into feat/bitbucket-mend-dashboard-check-ext

RahulGautamSingh commented on lib/modules/platform/bitbucket/index.spec.ts:
```diff
@@ -258,6 +258,244 @@ describe('modules/platform/bitbucket/index', () => {
+      expect(postRequests).toHaveLength(0);
+    });
+
+    it('enabled: should log debug message when creating new status', async () => {
```
Can't this log message be tested in the second test itself?

RahulGautamSingh reviewed (commented):
LGTM

zharinov committed:
Remove redundant logging test case

The test case for debug logging was redundant since the logging
assertions were already covered in other test cases.

zharinov committed:
Merge branch 'main' into feat/bitbucket-mend-dashboard-check-ext

RahulGautamSingh commented on lib/modules/platform/bitbucket/index.ts:
```diff
@@ -230,6 +231,27 @@ export async function initRepo({
 
     config.defaultBranch = mainBranch;
 
+    if (bbMendAppDashboardStatus) {
```
I don't think this be available here, if set in repo config? The option should be self-hosted

zharinov committed:
Merge branch 'main' into feat/bitbucket-mend-dashboard-check-ext

zharinov committed:
Make bbMendAppDashboardStatus self-hosted

Add bbMendAppDashboardStatus to inherited config options

zharinov committed:
Move bbMendAppDashboardStatus to self-hosted config docs

zharinov committed:
Merge branch 'main' into feat/bitbucket-mend-dashboard-check-ext

zharinov committed:
Merge branch 'main' into feat/bitbucket-mend-dashboard-check-ext

zharinov committed:
Fix: Add eslint-disable for vitest rule

zharinov committed:
Merge branch 'main' into feat/bitbucket-mend-dashboard-check-ext

viceice reviewed (commented):
otherwise LGTM

zharinov committed:
Merge branch 'main' into feat/bitbucket-mend-dashboard-check-ext

# Conflicts:
#	lib/modules/platform/bitbucket/index.ts
#	lib/modules/platform/types.ts

zharinov committed:
Merge branch 'feat/bitbucket-mend-dashboard-check-ext' of github.com:zharinov/renovate into feat/bitbucket-mend-dashboard-check-ext

zharinov committed:
Remove unused snapshot

zharinov commented on lib/modules/platform/bitbucket/__snapshots__/index.spec.ts.snap:
```diff
@@ -147,3 +147,5 @@ exports[`modules/platform/bitbucket/index > massageMarkdown() > returns diff fil
 
 exports[`modules/platform/bitbucket/index > updatePr() > throws an error on failure to get current list of reviewers 1`] = `[HTTPError: Response code 500 (Internal Server Error)]`;
+
+exports[`modules/platform/bitbucket/index ensureComment() does not throw 1`] = `false`;
```
No idea, deleted it

zharinov committed:
Merge branch 'main' into feat/bitbucket-mend-dashboard-check-ext

zharinov committed:
Fix lint

zharinov committed:
Remove unused `ignorePrAuthor` parameter

zharinov committed:
Use GlobalConfig

zharinov committed:
Update global.ts

RahulGautamSingh reviewed (commented):
Else, lgtm

viceice reviewed (commented):
conflicted
